### PR TITLE
Add hardcover as a service

### PR DESCRIPTION
**Pull request by pietbrauer**
*The other one became to messy.*

Hardcover is a new awesome in-house alternative to coveralls. We have been using it for the past 3 months internally and finally open sourced it.

For it to succeed we need this little PR merged into it.

You can find detailed information on how it works here: https://github.com/xing/hardcover#slather

Thanks for the good work, slather was a key for this project to work.

:dancer:

### CODE REVIEW

**Comment by marklarr** [comment]
Hey Piet, Hardcover looks amazing! Can't wait to start using it myself. I'll have some time to look over this PR during the work day tomorrow. In the meantime, it looks like the build is failing on Travis due to a couple of failing specs :)

**Comment by pietbrauer** [comment]
Hey @marklarr, thanks for the kind words! We also have a demo now: https://damp-shore-4827.herokuapp.com

I also removed some tests that were not providing value, so you guys could have a look if you fancy?

File: lib/slather/coverage_service/hardcover.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
+module Slather
+  module CoverageService
+    module Hardcover
+
+      def yml_filename
+        '.hardcover.yml'
```
Comment (marklarr) [inline lib/slather/coverage_service/hardcover.rb]: I'm thinking maybe we can just add this values to the `.slather.yml` so they only need one? Like we do for the ci_access_token https://github.com/xing/slather/blob/new_hardcover/lib/slather/project.rb#L107 (on a side note, it looks like I could probably DRY up those `_from_yml` methods to have less duplicate code)

**Comment by marklarr** [comment]
I'm realizing that there is a lot of duplicate code between the `hardcover` and `coveralls` classes, and I want to do some refactoring to allow polymorphism from a common base class there. However, I really want to help get Hardcover up and running, so let's merge this, and I'll refactor this weekend.

If you can just move the configuration values from `.hardcover.yml` to `.slather.yml`, then I'll merge, people can use hardcover, and I'll refactor without any breaking changes. Sound good?

**Comment by pietbrauer** [comment]
Yep, you are absolutely right. We designed hardcovers API to be as close as possible to the one provided by coveralls to make use of existing tooling.

I can make the changes quickly tomorrow but feel free to start your refactoring. I don't think someone will go ahead and deploy everything on the weekend already ;).

**Comment by marklarr** [comment]
Hey @pietbrauer -- as soon as you can make the `yml` chance, I can merge this!

**Comment by pietbrauer** [comment]
@marklarr Sorry for the delay, totally lost it in my mails! But I've added tests and refactored a bit. Ready for review now.

**Comment by pietbrauer** [comment]
Ping? :relaxed:

**Comment by marklarr** [comment]
Awesome! Thanks :)

**Comment by ajimix** [comment]
Will be nice to do a release with this :)

**Pull request merged**